Zend/zend_system_id.c

```diff
@@ -0,0 +1,79 @@
+/*
+   +----------------------------------------------------------------------+
+   | Copyright (c) The PHP Group                                          |
+   +----------------------------------------------------------------------+
+   | This source file is subject to version 3.01 of the PHP license,      |
+   | that is bundled with this package in the file LICENSE, and is        |
+   | available through the world-wide-web at the following url:           |
+   | http://www.php.net/license/3_01.txt                                  |
+   | If you did not receive a copy of the PHP license and are unable to   |
+   | obtain it through the world-wide-web, please send a note to          |
+   | license@php.net so we can mail you a copy immediately.               |
+   +----------------------------------------------------------------------+
+   | Authors: Sammy Kaye Powers <sammyk@php.net>                          |
+   |          Dmitry Stogov <dmitry@php.net>                              |
+   +----------------------------------------------------------------------+
+ */
+
+#include "php.h"
+#include "zend_system_id.h"
+#include "zend_extensions.h"
+#include "ext/standard/md5.h"
+#include "ext/hash/php_hash.h"
+
+ZEND_API char zend_system_id[32];
+
+static PHP_MD5_CTX context;
+static int finalized = 0;
+
+ZEND_API void zend_add_system_entropy(const char *module_name, const char *hook_name, const void *data, size_t size)
+{
+	if (finalized == 0) {
```

Comment:
It may make sense to inform about changing something after finalization. I'm not sure...

Reply:
Good point. I'll update it to return a `ZEND_RESULT_CODE`.

Reply:
I'd prefer if this was a `ZEND_ASSERT(!finalized)` instead. Seems unlikely anyone is going to check the return value.

Reply:
By design the system ID is finalized right before the `zend_post_startup_cb` hook. Adding `ZEND_ASSERT(!finalized)` will fail for any engine hooks added in a `zend_post_startup_cb` hook (like OPcache does). There are workarounds to add entropy for late hooks (listed in the "Caveats" section of the PR description), but it looks like this design requirement locks us into using the result code approach here.